src/common/survival_util.h

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright 2019 by Contributors
+ * \file survival_util.h
+ * \brief Utility functions, useful for implementing objective and metric functions for survival
+ *        analysis
+ * \author Avinash Barnwal, Hyunsu Cho and Toby Hocking
+ */
+#ifndef XGBOOST_COMMON_SURVIVAL_UTIL_H_
+#define XGBOOST_COMMON_SURVIVAL_UTIL_H_
+
+#include <xgboost/enum_class_param.h>
+#include <memory>
+
+namespace xgboost {
+namespace common {
+
+// Choice of distribution for the noise term in AFT
+enum class AFTDistributionType : int {
+  kNormal = 0, kLogistic = 1, kExtreme = 2
+};
+
+}  // namespace common
+}  // namespace xgboost
+
+DECLARE_FIELD_ENUM_CLASS(xgboost::common::AFTDistributionType);
+
+namespace xgboost {
+namespace common {
+
+// Constant PI
+const double kPI = 3.14159265358979323846;
+
+struct AFTParam : public dmlc::Parameter<AFTParam> {
+  AFTDistributionType aft_noise_distribution;
+  float aft_sigma;
```

Comment:
This is a personal preference, `gamma` looks good on paper, but in code it doesn't ...  I would go with something more specific like `distribution_scale_factor` (still not a good name) and document it with the `gamma` parameter used in xxx paper.

Reply:
Did you mean aft_sigma?

Reply:
I went ahead and chose the new names `aft_loss_distribution` and `aft_loss_distribution_scale`.